src/cosmos/engine/CosmosEngine.ts

```diff
@@ -176,38 +177,73 @@ export class CosmosEngine extends CurrencyEngine<
 
             const { assets, memo, metadata } = params
 
+            const memos: EdgeMemo[] = []
+            if (memo !== '') {
+              memos.push({ type: 'text', value: memo })
+            }
+
             const msg = this.tools.methods.deposit({
               assets,
               memo,
               signer: this.walletInfo.keys.bech32Address
             })
             const unsignedTxHex = this.createUnsignedTxHex([msg], memo)
 
-            const { gasFeeCoin, gasLimit, networkFee } =
-              await this.calculateFee({
-                messages: [msg],
-                memo
-              })
+            let { gasFeeCoin, gasLimit, networkFee } = await this.calculateFee({
+              messages: [msg],
+              memo
+            })
             const otherParams: CosmosTxOtherParams = {
               gasFeeCoin,
               gasLimit,
               unsignedTxHex
             }
 
+            let currencyCode = this.currencyInfo.currencyCode
+            let tokenId = null
+            let nativeAmount = `-${networkFee}`
+            let parentNetworkFee: string | undefined
+            // For one asset, we can make a pretty good guess at filling in the EdgeTransaction fields
+            if (assets.length === 1) {
+              const asset = assets[0]
+
+              currencyCode = assetFromString(asset.asset).ticker
+              if (currencyCode !== this.currencyInfo.currencyCode) {
+                const tokenObj = Object.entries(this.allTokensMap).find(
+                  ([_, token]) => {
+                    return token.currencyCode === currencyCode
+                  }
+                )
+                if (tokenObj != null) {
+                  tokenId = tokenObj[0]
+                } else {
+                  // if we don't know the asset, just let it continue as tokenId: null until we see it off the network.
+                }
+                nativeAmount = asset.amount
```

Comment:
should fallback to old implementation if you can't find the tokenId so the amount and tokenId are in the same units